client/src/components/Forms/CourseTaskSelect.tsx

```diff
@@ -1,20 +1,55 @@
 import * as React from 'react';
 import { Select, Form } from 'antd';
+import _ from 'lodash';
+
 import { CourseTask } from 'services/course';
 
-type Props = { data: CourseTask[]; onChange?: (id: number) => void };
+type Props = {
+  data: CourseTask[];
+  groupBy: string;
+  onChange?: (id: number) => void;
+};
+
+const MAX_DATE = '+275760-09-13T00:00:00.000Z';
 
 export function CourseTaskSelect(props: Props) {
-  const { data, onChange, ...rest } = props;
-  const selectProps = onChange ? { onChange } : {};
+  const { data, groupBy, onChange, ...options } = props;
+  const sortedData = data.sort(function (firstTask, secondTask) {
```

Comment:
minor but also it would be gret if you can convert it to arrow function. We prefer to use it inside array methods.

Reply:
Done